tests/callbacks_test.cpp

```diff
@@ -562,3 +562,66 @@ TEST_CASE("TimerStopCallbackTest", "[CallbacksTest]")
   // Add some time to account for the function to return.
   REQUIRE(timer.toc() < 2);
 }
+
+/**
+ * Make sure the ProgressBar callback will show the progress on the specified
+ * output stream if the MaxIterations parameter of the optimizer is 0.
+ */
+TEST_CASE("ProgressBarCallbackNoMaxIterationsTest", "[CallbacksTest]")
+{
+  SGDTestFunction f;
+  arma::mat coordinates = f.GetInitialPoint();
+
+  StandardSGD s(0.0003, 1, 0, DBL_MAX, true);
+
+  std::stringstream stream;
+  s.Optimize(f, coordinates, ProgressBar(10, stream));
+
+  REQUIRE(stream.str().length() > 0);
+}
+
+/**
+ * Make sure the ProgressBar callback will show the progress on the specified
+ * output stream with the correct epoch number if the MaxIterations parameter
+ * of the optimizer is 0.
+ */
+TEST_CASE("ProgressBarCallbackNoMaxIterationsEpochTest", "[CallbacksTest]")
+{
+  SGDTestFunction f;
+  arma::mat coordinates = f.GetInitialPoint();
+
+  StandardSGD s(0.0003, 1, 0, DBL_MAX, true);
+
+  std::stringstream stream;
+  s.Optimize(f, coordinates, ProgressBar(10, stream));
+  bool success;
+  if (stream.str().find("Epoch 1") != std::string::npos)
+  {
+    success = true;
+  }
+
+  REQUIRE(success == true);
```

Comment:
```suggestion
  REQUIRE(stream.str().find("Epoch 1") != std::string::npos);
```
Maybe this is cleaner?

Also we could add a simple test to ensure that the `/` isn't printed:

```
REQUIRE(stream.str().find("Epoch 1/") == std::string::npos);
```